File: mobile/lib/models/reminder.dart
class PetReminder{
  String? petRemID;
  String? petID;
  String note;
  String? nextRemDate;
  String? nextRemTime;



  PetReminder(this.petID,this.note,this.nextRemDate,this.nextRemTime);
  PetReminder.fromrems(this.petID,this.note,this.nextRemDate,this.petRemID,this.nextRemTime);

  PetReminder.fromJson(Map<String, dynamic> json)
      : petID = json['petID'].toString(),
        note = json['note'],
        nextRemDate = json['nextRemDate'].toString(),
        nextRemTime = json['nextRemTime'].toString(),
        petRemID =json['petRemID'].toString();



  Map toJson() => {
    'petID': petID,
    'note': note,
    'nextRemDate' : nextRemDate,
    'nextRemTime' : nextRemTime
  };
}

File: server/models/reminder.js
module.exports = (sequelize, DataTypes) => {
    const PetReminders = sequelize.define("PetReminders", {
      petRemID : {
        type: DataTypes.INTEGER,
        autoIncrement: true,
        primaryKey: true
      },
      petID: {
        type: DataTypes.INTEGER,
        allowNull: false,
      },
      
      note: {
        type: DataTypes.STRING,
        allowNull: false,
      },
      nextRemDate: {
        type: DataTypes.DATEONLY,
        allowNull: false,
      },
      nextRemTime: {
        type: DataTypes.TIME,
        allowNull: false,
      },
      
      

    },{

      timestamps: false,
      freezeTableName: true
    }
    );
  
    PetReminders.associate = (models) => {
        PetReminders.belongsTo(models.Pets, {
          foreignKey: 'petID',
          onDelete: "cascade",
          onUpdate: "cascade",
        });
        
      };
    return PetReminders;
  };
  